plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/util/ConstantPoolHelpers.kt

```diff
@@ -0,0 +1,86 @@
+package io.sentry.android.gradle.instrumentation.util
+
+import java.lang.reflect.Field
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.ClassWriter
+
+/**
+ * Looks up for the original [ClassWriter] up the visitor chain by looking at the private `cv` field
+ * of the [ClassVisitor].
+ */
+internal fun ClassVisitor.findClassWriter(): ClassWriter? {
+    var classWriter: ClassVisitor = this
+    while (!ClassWriter::class.java.isAssignableFrom(classWriter::class.java)) {
+        val cvField: Field = try {
+            classWriter::class.java.allFields.find { it.name == "cv" } ?: return null
+        } catch (e: Throwable) {
+            return null
+        }
+        cvField.isAccessible = true
+        classWriter = (cvField.get(classWriter) as? ClassVisitor) ?: return null
+    }
+    return classWriter as ClassWriter
+}
+
+/**
+ * Looks up for [ClassReader] of the [ClassWriter] through intermediate SymbolTable field.
+ */
+internal fun ClassWriter.findClassReader(): ClassReader? {
+    val clazz: Class<out ClassWriter> = this::class.java
+    val symbolTableField: Field = try {
+        clazz.allFields.find { it.name == "symbolTable" } ?: return null
+    } catch (e: Throwable) {
+        return null
+    }
+    symbolTableField.isAccessible = true
+    val symbolTable = symbolTableField.get(this)
+    val classReaderField: Field = try {
+        symbolTable::class.java.getDeclaredField("sourceClassReader")
+    } catch (e: Throwable) {
+        return null
+    }
+    classReaderField.isAccessible = true
+    return (classReaderField.get(symbolTable) as? ClassReader)
+}
+
+/**
+ * Looks at the constant pool entries and searches for R8 markers
+ */
+internal fun ClassReader.isMinifiedClass(): Boolean {
+    val charBuffer = CharArray(maxStringLength)
+    // R8 marker is usually in the first 3-5 entries, so we limit it at 10 to speed it up
+    // (constant pool size can be huge otherwise)
+    val poolSize = minOf(10, itemCount)
+    for (i in 1 until poolSize) {
+        try {
+            val constantPoolEntry = readConst(i, charBuffer)
+            if (constantPoolEntry is String && "~~R8" in constantPoolEntry) {
+                // ~~R8 is a marker in the class' constant pool, which r8 itself is looking at when
+                // parsing a .class file. See here -> https://r8.googlesource.com/r8/+/refs/heads/main/src/main/java/com/android/tools/r8/dex/Marker.java#53
```

Comment:
What if its minified by proguard or dexguard?

Reply:
dexguard only works for android apps afaik, so not possible to minify libs. + Our license has expired, so I have no way to test it :)

just checked with proguard - there's no marker or anything in the constant pool saying it was minified with proguard. 

But I'll double-check with the guardsquare guys, I think I can reach them somewhere.

Reply:
@romtsn thanks.
So if you are using proguard, do the unit tests break with minified libs?

Reply:
so far - nope, I think `gms` and `google-play-services` are still minified with proguard and not R8 (because they also don't have any marker in there), but for those our fixes are working fine so far.

I think going forward we should still introduce the `excludes` option so people can manually excluded those classes that fail without us needing to bugfix everytime. I think with this change we cover a good portion of minified libs already